src/forms/password/Password.tsx

```diff
@@ -0,0 +1,141 @@
+    errorHelperText?: string; // shown when state isError is true
+    placeholder?: string;
+    label?: string;
+    onChange?: (evt: React.ChangeEvent<HTMLInputElement>) => void;
+    onBlur?: (evt: React.FocusEvent<HTMLInputElement>) => void;
+    name: string;
+    id?: string;
+    value?: string;
+    defaultValue?: string;
+    maxPasswordLength?: number;
+    minPasswordLength?: number;
+    required?: boolean; // auto-check on blur if there's a value
+    error?: boolean; // force error state of component
+    showPassword?: boolean;
+};
+
+type PasswordState = {
+    show: boolean;
+    type: string;
+};
+export class Password extends React.PureComponent<PasswordProps, PasswordState> {
+    static defaultProps = {
+        showPassword: true,
+    };
+
+    state: PasswordState = {
+        show: false,
+        type: "password",
+    };
+
+    private showHidePassword = () => {
+        this.setState(prevState => ({
+            show: !prevState.show,
+            type: !prevState.show ? "text" : "password",
+        }));
+    };
+
+    private static renderHelperText(helperText: ReactNode): ReactNode {
+        return <span className="clr-subtext">{helperText}</span>;
+    }
+
+    private static renderLabel(label: string) {
+        return <label className="clr-control-label clr-col-12 clr-col-md-2">{label}</label>;
+    }
+
+    render() {
+        const {
+            className, //prettier
+            disabled,
+            helperText,
+            label,
+            value,
+            defaultValue,
+            errorHelperText,
+            error,
+            style,
+            name,
+            required,
+            id,
+            maxPasswordLength,
+            minPasswordLength,
+            placeholder,
+            showPassword,
+        } = this.props;
+
+        const {show, type} = this.state;
+
+        let classNames = [
+            "clr-control-container",
+            error && "clr-error",
+            label && "clr-col-md-10 clr-col-12",
+            className,
+        ];
+
+        if (disabled) classNames.push("clr-form-control-disabled");
+        return (
+            <div className="clr-form clr-form-horizontal ng-pristine ng-valid ng-touched">
+                <div className={utils.classNames(["clr-form-control", label && "clr-row"])}>
+                    {label && Password.renderLabel(label)}
+                    <div className={utils.classNames(classNames)} style={{width: "100%"}}>
+                        <div className="clr-input-wrapper">
+                            <div className="clr-input-group clr-row" style={{...style, paddingRight: "1rem"}}>
+                                <input
+                                    maxLength={maxPasswordLength}
+                                    minLength={minPasswordLength}
+                                    name={name}
+                                    defaultValue={defaultValue}
+                                    value={value}
```

Comment:
When we enter the value in the password component, and if I do not have a onChange which sets the value prop explicitly, how will I be able to get the password value? I think value should be part of the password state

Reply:
done